### DISCUSSION TITLE
Reactive initialPageParam on Inifinite Queries

### DISCUSSION THREAD

**Discussion by MarioC3** [TanStack/query#6375] [Q&A]
Is there a way to make the `initialPageParam` attribute reactive to change the initial parameters if one of the parameters changes?

I'm trying to do something like this:
```
useInfiniteQuery({
		queryKey: ['products', key, { filters }],
		queryFn: ({ pageParam }) => getProducts(pageParam),
		initialPageParam: {
			key,
			offset: offset?.value,
			limit: limit?.value,
			filters: filters?.value,
		},
		getNextPageParam: (lastPage) => {
			if ((offset?.value ?? 0) >= lastPage.count) return undefined
			if (lastPage.filteredCount === 0) return undefined
			if (lastPage.count <= (limit?.value ?? 0)) return undefined
			return {
				key,
				offset: offset?.value,
				limit: limit?.value,
				filters: filters?.value,
			}
		},
		placeholderData: keepPreviousData,
	})
```

Where `filters` is a reactive property, so every time it changes, TanStack Query picks it up and re-fetches the query. The problem is that initialPageParams remain the same, even if `filters` change, thus always passing the same value for `filters` and doing the same API call.

**Comment by TkDodo** [comment]
can you show this in a sandbox (preferably react)? `initialPageParam` should always be the latest value, and always considered when a new cache entry fetches for the first time
